### PR TITLE
[RA]Mousewheel sidebar scrolling 

### LINKED ISSUE #307: [Suggestion] Add scrollwheel support in sidebar
I don't know how people played this back in the day, my finger hurts :stuck_out_tongue:

### PR DESCRIPTION
**Pull request by mvdhout1992**
Implements handling of mousewheel scroll bar up and down SDL events and add sidebar scrolling code

### CODE REVIEW

File: common/settings.cpp [hunk 1]
```diff
@@ -21,6 +23,8 @@ void SettingsClass::Load(INIClass& ini)
     Video.Width = ini.Get_Int("Video", "Width", Video.Width);
     Video.Height = ini.Get_Int("Video", "Height", Video.Height);
     Video.FrameLimit = ini.Get_Int("Video", "FrameLimit", Video.FrameLimit);
+
+    Options.MouseWheelScrolling = ini.Get_Bool("Options", "MouseWheelScrolling", Video.FrameLimit);
```
Comment (hifi) [inline common/settings.cpp:27]: Default value is wrong.

File: common/settings.cpp [hunk 2]
```diff
@@ -31,4 +35,6 @@ void SettingsClass::Save(INIClass& ini)
     ini.Put_Int("Video", "Width", Video.Width);
     ini.Put_Int("Video", "Height", Video.Height);
     ini.Put_Int("Video", "FrameLimit", Video.FrameLimit);
+
+    ini.Put_Bool("Options", "MouseWheelScrolling", Video.FrameLimit);
```
Comment (hifi) [inline common/settings.cpp:39]: Writes the wrong value to ini.

**Review by hifi — CHANGES_REQUESTED** [review]
Some copypasta issues but otherwise looks okay. And please add TD support in the same PR so they keep feature parity.

Also these changes are small enough to fit a single commit even when hooking up both games. Cleanup commits should be squashed away anyway so that's something that needs to be done regardless of how the commits are laid out.

File: common/settings.cpp (hunk 2 above)
Comment (mvdhout1992) [reply common/settings.cpp:39]: Will fix the INI writing.

Comment (mvdhout1992) [reply common/settings.cpp:39]: Ok I'll add TD support too and fix the ini copypasta.

**Comment by OmniBlade** [comment]
Superceded by cleaned up version #710.

**Pull request closed without merging**